File: Level2/큰 수 만들기.py
number = "4177252841"
k = 4

def solution(number, k):
    lis= []
    for i in number:
        while len(lis) > 0 and k > 0 and int(lis[-1]) < int(i):
            k -= 1
            lis.pop()
        lis.append(i)
    answer = ''.join(lis)
    return answer[:len(number)-k]


did = solution(number, k)
print(did)

File: Level2/큰 수 만들기.java
import java.util.*;
class Solution {
    public String solution(String number, int k) {
        int tmp = k;
        Stack<Character> stack = new Stack<>();
        for(int i=0; i<number.length(); i++) {
            int now = number.charAt(i);
            while(!stack.isEmpty() && stack.peek() < now && k > 0) {
                stack.pop();
                k--;
            }
            stack.push(number.charAt(i));
        }
        StringBuilder sb = new StringBuilder();
        for(int i=0; i<number.length()-tmp; i++)
            sb.append(stack.get(i));
        return sb.toString();
    }
}
